easy_phi/auth.py

```diff
@@ -262,24 +267,143 @@ def get(self):
+            _user_cache = sorted([user for user in users if user])
+        return _user_cache
+
+    @staticmethod
+    def set_users(users):
+        global _user_cache
+        try:
+            fh = open(options.security_password_auth_user_list_path, 'w')
+            fh.write("\n".join(sorted(users)))
+        except IOError:
+            return False
+        fh.close()
+        _user_cache = None
+        return True
+
+    @staticmethod
+    def check_password(username, password):
+        """ Check if user is registered user and password matches stored one """
+        users = PasswordAuthAPIHandler.get_users() or []
+        return username in users \
+            and password == keyring.get_password(service, username)
+
+    @http_basic(admin_auth)
+    def prepare(self):
+        self.set_header('Content-Type', 'text/plain')
+        users = PasswordAuthAPIHandler.get_users()
+        if users is None:
+            self.set_status(403)
+            self.finish('Password authorization is not configured properly. '
+                        'Please check system documentation.')
+            return
+
+        if self.request.method in ('POST', 'DELETE', 'HEAD'):
+            user = self.get_argument('user')
+            if user not in users:
+                self.set_status(404)
+                self.finish('User {user} not found'.format(user=user))
+                return
+            self.user = user
+
+        if self.request.method in ('POST', 'PUT'):
+            # "or ''" is to handle PUT/POST without body
+            password = self.request.body or ''
+            if not 5 < len(password) < 31 or not re.search("\d", password) \
+                    or not re.search("[a-zA-Z]", password):
+                self.set_status(400)
+                self.finish('Password is mission or does not mathch safety '
+                            'criteria. Password should be 6 to 30 characters '
+                            'long, contain at least one digit and at least one '
+                            'letter.')
+                return
+            self.password = password
 
-def _check_password_from_file(username, password):
-    """ Utility function for password security backend
-    Workflow:
-        get salt from file
-        calculate password hash using salt
-        compare
-    WARNING: for security reasons hashes in file should be different from tokens
-    """
-    # TODO: implement this method
-    return True
+    def get(self):
+        """List users (ajax) or render passwords management page (otherwise)"""
+        users = PasswordAuthAPIHandler.get_users()
+        # Ajax
+        if self.request.headers.get('X-Requested-With') == 'XMLHttpRequest':
+            # i.e. it is ajax request
+            self.finish("\n".join(users))
+            return
+        # Non- Ajax
+        self.clear()  # reset content-type
+        self.render('admin_manage_passwords.html', users=users)
```

Comment:
content of this template will be moved to admin console when configuration wizard is implemented